### DISCUSSION TITLE
Lag and stationary object sticking

### DISCUSSION THREAD

**Discussion by pocipocster** [blakeblackshear/frigate#9394] [General]
Hi

I have 2 issues .
A few cameras have maybe a 10 second lag - which of course causes issues with for example automation of a light turning on in home assistant .
The 2nd issue which I thought was fixed is that stationary objects ( marked as a person ) ‘ stick ‘ and permanently stay - though in theory should be cleared after a time .

Frigate 0.13

mqtt:
  host: 192.168.8.126
  user: broker
  password: password
  
detectors:
  coral:
    type: edgetpu
    device: usb
    
ffmpeg:
  hwaccel_args: preset-vaapi
  #output_args:
     #record: preset-record-generic-audio-aac
    
go2rtc:
  streams:
    front:
      - rtsp://admin:xxxx@192.168.8.123/streaming/channels/102
      
    lounge:
      - rtsp://admin:xxxx@192.168.8.24/streaming/channels/102
      
    kitchen:
      - rtsp://admin:xxxx@192.168.8.25/streaming/channels/102
      
    door:
      - rtsp://admin:xxxxxx@192.168.8.177:554/h264Preview_01_sub
      
    kitchencorridor:
      - rtsp://admin:xxxxx@192.168.8.22:554/h264Preview_01_sub
  
cameras:
  front:
        ffmpeg:
            inputs:
              - path: rtsp://admin:xxxxx@192.168.8.123/streaming/channels/102
                roles:
                  - detect
                  - record
                  
        record: 
          enabled: True
          retain:
            days: 1
          events:
            retain:
              default: 21
              mode: active_objects
              objects:
                person: 21
                
        motion:
          mask: "640,0,640,65,0,96,0,0"
              
  lounge:
        ffmpeg:
            inputs:
              - path: rtsp://admin:xxxxxx@192.168.8.24/streaming/channels/102
                roles:
                  - detect
                  
  kitchen:
        ffmpeg:
            inputs:
              - path: rtsp://admin:xxxxxxx@192.168.8.25/streaming/channels/102
                roles:
                  - detect
                  
        objects:
          filters:
            person:
              threshold: 0.75
              
  kitchencorridor:
        ffmpeg:
            inputs:
              - path: rtsp://admin:xxxxx@192.168.8.22:554/h264Preview_01_sub
                input_args: preset-rtsp-restream
                roles:
                  - detect
                 
        objects:
          filters:
            person:
              threshold: 0.75
                  
  door:
        ffmpeg:
            inputs:
              - path: rtsp://admin:xxxxxx@192.168.8.177/h264Preview_01_sub
                input_args: preset-rtsp-restream
                roles:
                  - detect
                  - record
                  
        record:
          enabled: True
          retain:
            days: 1
          events:
            retain:
              default: 21
              mode: active_objects
              objects:
                person: 21
                  
  #gate:
        #ffmpeg:
            #inputs:
              #- path: rtsp//admin:xxxxxxx@192.168.8.26/live/0/MAIN
                #roles:
                  #- detect
                  
            
        detect:
             enabled: True
             width: 640
             height: 380
             fps: 5
             
        objects:
          filters:
            person:
              threshold: 0.8
                 
        #record:
             #enabled: True
                 
        #filters:
          #person:
            #max_area: 100000

**Comment by NickM-27** [comment]
Where are you seeing this lag? What type of view are you using?

Can you shown an example of this stationary object? What does the debug live view show while this is happening?

**Reply by pocipocster** [reply]
Hi

The lag happens in real-time view . So I can stand infront a camera ( door or front ) and see the delay hence automations are triggered late ( say off a person ) . Jam peg or all views ( what view is used for motion / object detection? ) .

The ‘stuck ‘ object is more tricky as intermittent. But it acknowledges an object as a person ( though not a person ) and it stays permanently until the view is changed I.e I walk into the camera view .
I thought stationary objects had a ‘life span’ and then ended ( fps x 25 ?? - something like that )

**Reply by NickM-27** [reply]
1. So you're saying jsmpeg is delayed?

2. No, stationary objects don't have a lifespan, that's the point of stationary object tracking - to know about, track, and not be confused by objects that are stationary 

Sounds like in this case you need to follow the docs on reducing false positives.

**Reply by pocipocster** [reply]
Hi

yes jsmpeg is delayed .

ok - I am confused now about stationary objects . So a stationary objects remains so until it moves / changes ? . But the stationary object in question is a sofa - so I can’t mask it out I.e it’s flagged as a person .

**Reply by NickM-27** [reply]
What inference times do you see on the google coral?

You could likely use a max_area filter for a soda to filter it out.

**Reply by pocipocster** [reply]
Inference is good 9ms or so …
The delay ( need more testing ) only appears to happen on door and front camera I.e cameras that are recording ( not sure if that’s relevant ) .
I’ll look into max_area filter to reduce false positives .
